velox/vector/SimpleVector.h

```diff
@@ -307,17 +311,42 @@ class SimpleVector : public BaseVector {
   template <typename U = T>
   typename std::enable_if_t<std::is_same_v<U, StringView>, void> setAllIsAscii(
       bool ascii) {
-    ensureIsAsciiCapacity(length_);
+    ensureIsAsciiCapacity();
     asciiInfo.isAllAscii() = ascii;
     asciiInfo.asciiSetRows().setAll();
   }
 
+  template <typename U = T>
+  typename std::enable_if_t<std::is_same_v<U, StringView>, bool> getAllIsAscii()
+      const {
+    return asciiInfo.isAllAscii();
+  }
+
+  /// Provides const access to asciiInfo. Used for tests only.
+  template <typename U = T>
+  typename std::enable_if_t<std::is_same_v<U, StringView>, const AsciiInfo&>
+  testGetAsciiInfo() const {
```

Comment:
I think us leaking asciiInfo to tests is probably something we should avoid..
At the very least this should only be enabled in debug builds with the ifndef NDEBUG macro

Reply:
We already have a number of test methods in the Velox code.
It is a common practice.
Saying that, I don't like having test-only code in the production code myself, but sometimes we have no choice.

Reply:
One way to do this without leaking asciiInfo, is creating a udf that has an ascii fast path and ensure that is called the right way ?

Reply:
You mean instead of a small const accessor function you want to implement a test-only UDF and a test for it?
Can I object to it? :)
It seems like an excessive amount of trouble.

Also, the idea here is to have a test that would cover all use cases for a SimpleVector and ensure we've got our memory allocated.
Making a UDF + test that covers all that would be kind of tricky.
We got to move fast and don't break things. ;)